packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapsPlugin.java

```diff
@@ -181,5 +194,88 @@ public void onActivityDestroyed(Activity activity) {
     public Lifecycle getLifecycle() {
       return lifecycle;
     }
+
+    @NonNull
+    @Override
+    public SavedStateRegistry getSavedStateRegistry() {
+      return savedStateRegistryController.getSavedStateRegistry();
+    }
+  }
+
+  /**
+   * This logic is incredibly hacky, but is necessary to address multiple problems with the Flutter
```

Comment:
nit: please start with a summary of what it does, the disclosure can come second 😄

Reply:
Done